### PR TITLE
[improvement](mtmv)mtmv support partition by hms table

### PR DESCRIPTION
**Pull request by zddr**
## Proposed changes

MTMV support hive tables as related tables。

Add an interface: `MTMVRelatedTableIf.java` Provide some methods to support materialized view maintenance partitioning

`HivePartition` add parameters attribute, storage partition last update time, etc

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by zddr** [comment]
run buildall

**Comment by zddr** [comment]
run buildall

**Comment by zddr** [comment]
run buildall

**Comment by zddr** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/mtmv/MTMVRelatedTableIf.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * The table that implements this interface can serve as a partition table followed by MTMV
+ */
+public interface MTMVRelatedTableIf extends TableIf {
+
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/mtmv/MTMVRelatedTableIf.java:35]: Add comment for all these interface

File: fe/fe-core/src/main/java/org/apache/doris/catalog/OlapTable.java [hunk 2]
```diff
@@ -2535,4 +2537,37 @@ public List<Tablet> getAllTablets() throws AnalysisException {
...
+        return getPartitionInfo().getIdToItem(false);
+    }
+
+    @Override
+    public long getPartitionLastModifyTime(long partitionId, PartitionItem item) throws AnalysisException {
+        return getPartitionOrAnalysisException(partitionId).getVisibleVersionTimeIgnoreInit();
+    }
+
+    @Override
+    public long getLastModifyTime() {
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/OlapTable.java:2557]: what will the last modify time be if a partition is dropped?

File: fe/fe-core/src/main/java/org/apache/doris/planner/external/hudi/HudiScanNode.java [hunk 3]
```diff
@@ -194,7 +194,7 @@ private List<HivePartition> getPrunedPartitions(
                     return filteredPartitionIds.stream().map(id -> {
                         String path = basePath + "/" + partitionIdToNameMap.get(id);
                         return new HivePartition(
-                                dbName, tblName, false, inputFormat, path, partitionValuesMap.get(id));
+                                dbName, tblName, false, inputFormat, path, partitionValuesMap.get(id), null);
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/planner/external/hudi/HudiScanNode.java:197]: Better pass an empty map, not `null`

File: fe/fe-core/src/main/java/org/apache/doris/catalog/OlapTable.java (hunk 2 above)
Comment (zddr) [reply fe/fe-core/src/main/java/org/apache/doris/catalog/OlapTable.java:2557]: olap table has same problem,resolve together with other PRs

**Comment by zddr** [comment]
run buildall

**Comment by zddr** [comment]
run pipelinex_p0

**Review by morningman — APPROVED** [review]
LGTM

**Review by wuwenchi — APPROVED** [review]

**Pull request merged**